### PR TITLE
Release 1.1

### PR DESCRIPTION
**Pull request by fluffyfly**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -33,7 +33,7 @@
         <dependency>
         	<groupId>org.springframework.ldap</groupId>
         	<artifactId>spring-ldap-core</artifactId>
-        	<version>2.3.1.RELEASE</version>
+        	<version>2.3.1.RELEASE</version><!--$NO-MVN-MAN-VER$-->
```
Comment (fluffyfly) [inline pom.xml]: useless

File: src/main/java/org/example/api/endpoint/ApiController.java [hunk 2]
```diff
@@ -89,19 +93,19 @@ public TokenObject retrieve(
...
-    		) throws LockedTokenException {
+    		) throws LockedTokenException {    	
     	return tokenService.deleteToken(tok, false);
     }
     
     @RequestMapping(value = "/tokens/unlock", method = RequestMethod.POST, consumes = "application/json", produces = "application/json")
     @ApiOperation(value = "Unlock a token",
     		      notes = "Unlock a token for a specific user/channel/schema\n\n"
-    		    		  + "Possible return codes : a 404 if the token is not found or a 400 if input parameters are invalid",
+    		    		  + "Possible return codes : a 404 if the token is not found, a 400 if input parameters are invalid, or a 401 if the token is not locked",
```
Comment (fluffyfly) [inline src/main/java/org/example/api/endpoint/ApiController.java]: If the token is not locked, a 200 should be returned, not a 401.
401 Unauthorized indicates that the request is not authenticated, which is not the case here

File: src/main/java/org/example/api/model/UserInfo.java [hunk 3]
```diff
@@ -0,0 +1,176 @@
+package org.example.api.model;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public class UserInfo {
+
+	@JsonProperty("username") private String username;
```
Comment (fluffyfly) [inline src/main/java/org/example/api/model/UserInfo.java]: Maybe we don't need to declare all those attributes, only declare those that we will retrieve (see spec)

File: src/main/java/org/example/api/model/dao/OpenAMDAO.java [hunk 4]
```diff
@@ -1,27 +1,72 @@
...
-		// here should call openam /json/authenticate and handle the back and forth
+	public boolean authenticate(String login, String password, String scope) {
+	
+		MultiValueMap<String, String> headers = new LinkedMultiValueMap<String, String>();
+	
+		headers.add("Content-Type","application/json");
+		headers.add("X-OpenAM-Username", login);
+		headers.add("X-OpenAM-Password", password);
+
+		return restTemplate.postForObject(openAMConfig.getUserAuthenticationUrl(scope), new HttpEntity<String>(headers), AuthenticationResponse.class).getMessage().equals("Authentication Successful");
```
Comment (fluffyfly) [inline src/main/java/org/example/api/model/dao/OpenAMDAO.java]: Be careful here, .getMessage() may or may not return a null pointer, maybe we should do it in 2 steps

File: src/main/java/org/example/api/model/dao/OpenAMDAO.java [hunk 5]
```diff
@@ -1,27 +1,72 @@
...
+		adminTokenResponse = restTemplate.postForObject(openAMConfig.getAdminAuthenticationUrl(), new HttpEntity<String>(headers), AdminTokenResponse.class);
+		
+		headers.clear();
+		headers.add("Content-Type","application/json");
+		headers.add(openAMConfig.getCookieName(), adminTokenResponse.getTokenId());
+
+		userInfoResult = restTemplate.exchange(openAMConfig.getUserAttributesUrl(login,scope), HttpMethod.GET, new HttpEntity<String>(headers), RetrieveInfoResult.class).getBody();
+		
+		//To do : Map attributes
+		if(userInfoResult.getResultCount()==1){
```
Comment (fluffyfly) [inline src/main/java/org/example/api/model/dao/OpenAMDAO.java]: Here create a new model with a constructor to map everything

File: src/main/java/org/example/api/service/OpenAMService.java [hunk 6]
```diff
@@ -12,8 +19,27 @@
...
+	
+	public User authenticate(UserInput userInput) throws MultiResultException, InvalidAuthenticationException {
+		
+		User user = new User();
+		
+		if(!user.equals(null)){
+			if(amDAO.authenticate(userInput.getLogin(), userInput.getPassword(), userInput.getScope())){	
+				try{
+					user = amDAO.retrieveUserAttributes(userInput.getLogin(),userInput.getScope());
+					log.info("Timestamp : {}\t EventType : {}\tInputParameters : {}\tResult : {}\tSource : {} ", new SimpleDateFormat("yyyy.MM.dd "+"HH:mm:ss.SSS").format(new Date()),"Login/Password authentication", "[login : "+userInput.getLogin() +"\tScope : "+userInput.getScope()+"]", user, null);
```
Comment (fluffyfly) [inline src/main/java/org/example/api/service/OpenAMService.java]: Create a new model with all fields, and map with a method or a constrructor

File: src/main/java/org/example/api/service/TokenService.java [hunk 7]
```diff
@@ -29,41 +33,64 @@
...
     @Autowired
-    private SchemaListConfig ac;
+    private SchemaListConfig sc;
     
     @Autowired
     private SecureRandom secureRandom;
+    
+    private static Logger log = LoggerFactory.getLogger(TokenService.class);
+    
+    private String logConstructor(String timestamp, String eventType, String inputs, Integer result, String details, String source){
```
Comment (fluffyfly) [inline src/main/java/org/example/api/service/TokenService.java]: Same a previously, it should be a distinct model, not in the TokenService

File: src/main/java/org/example/api/model/dao/OpenAMDAO.java (hunk 5 above)
Comment (thomasraux) [inline src/main/java/org/example/api/model/dao/OpenAMDAO.java]: Isn't the purpose to check if we only get one result and then map the attributes into a User response ?

Comment (fluffyfly) [inline src/main/java/org/example/api/model/dao/OpenAMDAO.java]: Yes, you should just deport the code to map the RetrieveInfoResult <> User in the model instead

**Pull request merged**